File: scripts/parquet_to_json.py
import pyarrow.parquet as pq
import json

parquet_path = '/home/temp/_CODE/DPWH_2026_GAA/data/hierarchical_tree_leaf_nodes.parquet'
output_path = '/home/temp/_CODE/DPWH_2026_GAA/data/leaf_nodes.json'

print('Reading parquet file...')
table = pq.read_table(parquet_path)

print('Converting to dictionary...')
data = table.to_pydict()

print(f'Writing {len(next(iter(data.values())))} records to JSON...')
with open(output_path, 'w') as f:
    json.dump(data, f, indent=2)

print('Done!')

File: scripts/parquet_to_json.js
const parquet = require('parquetjs-lite');
const fs = require('fs');

async function convertParquetToJson() {
    const parquetPath = '/home/temp/_CODE/DPWH_2026_GAA/data/FY 2026_DPWH DETAILS ENROLLED COPY (Final)_leaf_nodes.parquet';
    const outputPath = '/home/temp/_CODE/DPWH_2026_GAA/data/leaf_nodes.json';

    console.log('Reading parquet file...');
    const reader = await parquet.ParquetReader.openFile(parquetPath);
    
    const cursor = reader.getCursor();
    let record = null;
    const records = [];
    
    while (record = await cursor.next()) {
        records.push(record);
    }
    
    await reader.close();
    
    console.log(`Writing ${records.length} records to JSON...`);
    fs.writeFileSync(outputPath, JSON.stringify(records, null, 2));
    console.log('Done!');
}

convertParquetToJson().catch(console.error);
